tests/tasks.py

```diff
@@ -293,3 +292,40 @@ def mytask(longer_arg):
             eq_(arg.names, ('longer-arg', 'l'))
             eq_(arg.attr_name, 'longer_arg')
             eq_(arg.name, 'longer_arg')
+
+        def decorated_function_names_are_preserved(self):
```

Comment:
Tests run with `spec`, and we have a local helper for it as well, see http://www.pyinvoke.org/development.html#running-management-tasks